File: Posts/Data-Strucutres-And-Algorithms/Prime-Number/Primality-Test.py
import math


def is_prime(x : int):
    if x <= 1:
        return False
    
    if x == 2 or x == 3:
        return True
    
    if x % 2 == 0:
        return False
    
    for i in range(3, math.ceil(math.sqrt(x)) + 1, 2):
        if x % i == 0:
            return False
    
    return True



p = 11
print(is_prime(p))

File: Posts/Data-Strucutres-And-Algorithms/Prime-Number/Primality-Test.cpp
#include <iostream>
#include <math.h>

using namespace std;

bool is_prime(int x) {
    if (x < 2) {
        return false;
    }
    if (x == 2 || x == 3) {
        return true;
    }
    if (x % 2 == 0) {
        return false;
    }
    for (int i = 3; i <= sqrt(x); i += 2) {
        if (x % i == 0) {
            return false;
        }
    }
    return true;
}

int main() {
    int p = 31;
    cout << is_prime(p) << endl;

    return 0;
}